### PR TITLE
refactor api links to come from enviroments

### PR DESCRIPTION
**Pull request by pallabez**
### Issue:
Fixes - #107

### Dev Tested:

- [x] Yes
- [ ] No

### Put the Feature behind Feature Flag

- [ ] Yes
- [x] No

### Follow-up Issues (if any):

> NOTE: After your PR is merged in main please verify its working on prod and raise a dev to main PR

### CODE REVIEW

File: nuxt.config.ts [hunk 1]
```diff
@@ -34,5 +34,12 @@ export default defineNuxtConfig({
       ['defineStore', 'definePiniaStore'], // import { defineStore as definePiniaStore } from 'pinia'
       ['useRepo', 'usePinaRepo']
     ]
+  },
+  runtimeConfig: {
+    public: {
+      goalsApi: process.env.NUXT_PUBLIC_GOAL_API,
+      rdsApi: process.env.NUXT_PUBLIC_RDS_API,
+      githubId: process.env.NUXT_PUBLIC_RDS_GITHUB_ID,
```
Comment (gauravsinhaweb) [inline nuxt.config.ts:42]: The URL key naming should be uppercase

**Review by gauravsinhaweb — CHANGES_REQUESTED** [review]

File: nuxt.config.ts (hunk 1 above)
Comment (pallabez) [reply nuxt.config.ts]: Done

File: adapters/goal.ts [hunk 2]
```diff
@@ -11,8 +11,10 @@ const goalSiteConfig = {
 }
 
 export const fetchGoals = async (): Promise<Goal[]> => {
+  const config = getConfig();
+
   const goals: Goal[] = await axios
-    .get(`${API.GOAL_BASE_URL}/v1/goal/`)
+    .get(`${config.GOALS_API}/v1/goal/`)
```
Comment (gauravsinhaweb) [inline adapters/goal.ts:17]: This URL naming is supposed to be the same as the URL either singular or plural depending on the incoming data.
Also verify API naming convention.

**Review by gauravsinhaweb — CHANGES_REQUESTED** [review]

File: adapters/goal.ts (hunk 2 above)
Comment (pallabez) [reply adapters/goal.ts:17]: I couldn't understand you. Do you mean it should be `.../v1/goals`?

Comment (gauravsinhaweb) [reply adapters/goal.ts:17]: Yes

Comment (pallabez) [reply adapters/goal.ts:17]: I don't think it is dependent on that unless I'm understand it wrong

**Review by gauravsinhaweb — APPROVED** [review]

**Pull request merged**